xdynamo/common_types.py

```diff
@@ -367,23 +369,28 @@ def _generate_dyn_keys(self) -> Set[DynKey]:
                             range_key=[range_combo[1], next_range[1]],
                             range_operator='between'
                         )
+                        dyn_keys.add(dyn_key)
                     else:
-                        dyn_key = DynKey(
-                            api=api,
-                            hash_key=hash_combo[1],
-                            range_key=range_combo[1],
-                            range_operator=range_operator
-                        )
-                    dyn_keys.add(dyn_key)
+                        # Don't add a hash-key if operator is not `eq`
+                        if hash_combo[1] == 'eq' and range_combo[1] == 'eq':
```

Comment:
@kupri-ar (no need to answer until after holidays):  I am wondering why we need to check to 'eq' here.  Can you provide more details?  This function is supposed to produce all possible ways to filter on the hash/range keys.  If we miss some of them, then we could get back unexpected results from dynamo when we get values based on the return of this function.

Perhaps there is an issue with non-eq keys that needs to be resolved elsewhere in the codebase?  Just would like some details and then I know what to suggest.